tools/python/boutiques/dataHandler.py

```diff
@@ -0,0 +1,276 @@
+#!/usr/bin/env python
+import os
+import requests
+import time
+import hashlib
+from boutiques import __file__ as bfile
+from boutiques.logger import raise_error, print_info
+from boutiques.util.utils import extractFileName, loadJson
+from boutiques.zenodoHelper import ZenodoHelper, ZenodoError
+
+
+class DataHandler(object):
+
+    # Constructor
+    def __init__(self):
+        self.cache_dir = getDataCacheDir()
+        self.cache_files = os.listdir(self.cache_dir)
+        self.descriptor_files = [fl for fl in self.cache_files
+                                 if fl.split('-')[0] == "descriptor"]
+        self.record_files = [fl for fl in self.cache_files
+                             if fl not in self.descriptor_files]
+
+    # Function to display the contents of the cache
+    # Option to display an example file which displays an pre-made example file
+    # or the first file of the cache if it exists
+    # Otherwise displays information about the cache and a list of files
+    def inspect(self, example=False):
+        self.example = example
+        # Display an example json file
+        if self.example:
+            # Display the first file in cache
+            if len(self.cache_files) > 0:
+                filename = self.cache_files[0]
+                file_path = os.path.join(self.cache_dir, filename)
+                self._display_file(file_path)
+            # Display the example file
+            else:
+                example_fn = os.path.join(bfile, "examples", "execution-data",
+                                          "execution-record.json")
+                self._display_file(example_fn)
```

Comment:
Not that it hurts but I'm not sure what is the rationale for this.

Reply:
Increase user information only, but optional.

Reply:
removed, directory access issues